File: 645-set-mismatch/solution.py
class Solution(object):
    def findErrorNums(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        hash_map = {v+1: 0 for v in range(len(nums))}
        for v in nums:
            hash_map[v] += 1
        
        res = [-1, -1]
        for k, v in hash_map.items():
            if v == 2:
                res[0] = k
            elif v == 0:
                res[1] = k
        
        return res

File: 645-set-mismatch/solution.cpp
class Solution {
public:
    vector<int> findErrorNums(vector<int>& nums) {
        vector<int> res; res.push_back(-1); res.push_back(-1);
        int dup = -1, missing = 1;
        for (int n : nums) {
            if (nums[abs(n)-1] < 0) dup = abs(n);
            else nums[abs(n)-1] *= -1;
        }
        for (int i = 0; i < nums.size(); i ++) {
            if (nums[i] > 0) {
                missing = i + 1;
                break;
            }
        }
        res[0] = dup; res[1] = missing;
        return res;
    }
};
